src/protocol.cpp

```diff
@@ -1544,6 +1559,37 @@ bool CProtocol::EvaluateVersionAndOSMes ( const CVector<uint8_t>& vecData )
     return false; // no error
 }
 
+void CProtocol::CreateRecorderStateMes ( const ERecorderState eRecorderState )
+{
+    CVector<uint8_t> vecData ( 1 ); // 1 byte of data
+    int              iPos = 0;      // init position pointer
+
+    // build data vector
+    // recorder state
+    PutValOnStream ( vecData, iPos, static_cast<uint32_t> ( eRecorderState ), 1 );
+
+    CreateAndSendMessage ( PROTMESSID_RECORDER_STATE, vecData );
+}
+
+bool CProtocol::EvaluateRecorderStateMes(const CVector<uint8_t>& vecData)
+{
+    int iPos = 0; // init position pointer
+
+    // check size
+    if ( vecData.Size() != 2 )
+    {
+        return true; // return error code
+    }
+
+    // server jam recorder state
+    const ERecorderState eRecorderState = static_cast<ERecorderState> ( GetValFromStream ( vecData, iPos, 1 ) );
```

Comment:
We should introduce a check for valid enum values. I don't know what the static_cast does, if no enum is defined for the received integer number.

Reply:
It's "undefined" -- i.e. the compiler is at liberty to do the easiest thing possible - just put the value into the storage "as is".

C++ hasn't got introspection so you can't check directly.

You'd have to have "known minimum" and "known maximum" values for each enum ("RS_VALID_MIN", "RS_VALID_MAX" etc) and then require that enum ranges not have intermediate _undefined_ values, I guess.

Maybe then have a macro that takes the "RS" (for example) and the result and checks for `xx_VALID_MIN` and `xx_VALID_MAX` using on the values passed.